src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1096,6 +1096,57 @@ bool is_object_required_superset(
     return true;
 }
 
+bool is_object_dependencies_superset(
+  json::Value const& older, json::Value const& newer) {
+    // "dependencies", if present, is a dict of <property, string_array |
+    // schema>. To be compatible, each key in older has to be in newer and the
+    // values have to be of the same type and compatible.
+    // older string array needs to be a subset of newer string array.
+    // older schema needs to be compatible with newer schema
+
+    auto [maybe_skip, older_p, newer_p] = extract_property_and_gate_check(
+      older, newer, "dependencies");
+    if (maybe_skip.has_value()) {
+        return maybe_skip.value();
+    }
+    // older and newer have "dependencies"
+
+    // helper to check a single dependency value
+    constexpr static auto dep_check =
+      [](json::Value const& o, json::Value const& n) -> bool {
+        if (o.IsArray() && n.IsArray()) {
+            // string array: n needs to be a a superset of o
+            // TODO: n^2 search
+            return std::ranges::all_of(
+              o.GetArray(), [n_array = n.GetArray()](json::Value const& p) {
+                  return std::ranges::find(n_array, p) != n_array.End();
+              });
+        }
+
+        if (o.IsObject() && n.IsObject()) {
+            // schemas: o and n needs to be compatible
+            return is_superset(o, n);
+        }
+
+        return false;
+    };
+
+    // all dependencies in older carry over in newer, and are compatible
+    // TODO: n^2 search
+    return std::ranges::all_of(
+      older_p->GetObject(),
+      [newer_dep = newer_p->GetObject()](auto const& older_dep) {
+          auto n_it = newer_dep.FindMember(older_dep.name);
+          if (n_it == newer_dep.MemberEnd()) {
+              // dependency definition removed, not compatible
+              return false;
+          }
+
+          // check that the dependency definition for key is compatible
+          return dep_check(older_dep.value, n_it->value);
+      });
```

Comment:
This shouldn't be static; it has no state. That makes it harder to capture into the lambda below. But I'm not sure factoring this out has added much clarity, I'd probably just inline this code.

```suggestion
    // all dependencies in older carry over in newer, and are compatible
    // TODO: n^2 search
    return std::ranges::all_of(
      older_p->GetObject(),
      [newer_dep = newer_p->GetObject()](auto const& older_dep) {
          auto n_it = newer_dep.FindMember(older_dep.name);
          if (n_it == newer_dep.MemberEnd()) {
              // dependency definition removed, not compatible
              return false;
          }

          auto const& o = older_dep.value;
          auto const& n = n_it->value;
          // check that the dependency definition for key is compatible
          if (o.IsArray() && n.IsArray()) {
              // string array: n needs to be a a superset of o
              // TODO: n^2 search
              return std::ranges::all_of(
                o.GetArray(), [n_array = n.GetArray()](json::Value const& p) {
                    return std::ranges::find(n_array, p) != n_array.End();
                });
          }

          if (o.IsObject() && n.IsObject()) {
              // schemas: o and n needs to be compatible
              return is_superset(o, n);
          }

          return false;
      });
```

Reply:
done